packages/quantic/force-app/main/default/lwc/headlessLoader/headlessLoader.js

```diff
@@ -0,0 +1,149 @@
+
+const DEBOUNCE_DELAY = 200;
+let timeout;
+
+/**
+ * Debounces a function execution.
+ * @param {Function} func The function for which to delay execution.
+ * @param {Number} wait The time to delay in milliseconds.
+ */
+const debounce = (func, wait) => {
+  return function executedFunction(...args) {
+    const later = () => {
+      clearTimeout(timeout);
+      func(...args);
+    };
+    clearTimeout(timeout);
+    // eslint-disable-next-line @lwc/lwc/no-async-operation
+    timeout = setTimeout(later, wait);
+  };
+};
+
+/**
+ * Cancels the delayed search query.
+ */
+const cancelInitialSearch = () => {
+  if (timeout) {
+    clearTimeout(timeout);
+    timeout = undefined;
+  }
+}
+
+/**
+ * Dispatches search request.
+ */
+const executeInitialSearch = debounce(() => {
+  window.coveoHeadless.engine.then((engine) => {
+    engine.dispatch(
+      CoveoHeadless.SearchActions.executeSearch(
+        CoveoHeadless.AnalyticsActions.logInterfaceLoad()
+      )
+    );
+  });
+}, DEBOUNCE_DELAY);
+
+/**
+ * Returns true if registered components are initialized, false otherwise.
+ */
+const areAllComponentsInitialized = () => !window.coveoHeadless.components.find(component => component.initialized === false);
+
+/**
+ * Returns the registered component object if it exists. 
+ * @param element 
+ */
+const getRegisteredComponent = (element) => window.coveoHeadless.components.find((component) => component.element === element);
+
+/**
+ * Loads dependencies and returns an initialized Headless engine. 
+ * @param element 
+ */
+async function initEngine(element) {
+  let engine;
+  try {
+    await loadScript(element, HeadlessPath + '/browser/headless.js');
+    await loadScript(element, AtomicPath + '/atomic-utils.js');
+  
+    const config = CoveoHeadless.HeadlessEngine.getSampleConfiguration();
+  
+    engine = new CoveoHeadless.HeadlessEngine({
+      configuration: config,
+      reducers: CoveoHeadless.searchAppReducers,
+    });
+  } catch (error) {
+    throw new Error('Fatal error: unable to initialize Coveo Headless: ' + error);
+  }
+  return engine;
+}
+
+/**
+ * Registers a component for future initialization.
+ * @param element 
+ */
+function registerComponentForInit(element) {
+  cancelInitialSearch();
+  if (!window.coveoHeadless) {
+    window.coveoHeadless = {
+      components: [],
+      engine: undefined
+    }
+  }
+  if (!getRegisteredComponent(element)) {
+    window.coveoHeadless.components.push({
+      element,
+      initialized: false
+    });
+  }
+}
+
+/**
+ * Sets registered component to initialized.
+ * @param element 
+ */
+function setComponentInitialized(element) {
+  const component = window.coveoHeadless ? getRegisteredComponent(element) : undefined;
+  if (!component) {
+    throw new Error('Fatal Error: Component was not registered before initialization');
+  }
+  component.initialized = true;
+  if (areAllComponentsInitialized()) {
+    executeInitialSearch();
```

Comment:
I could be wrong, but cancelling/forcing initial search from this loader feels out of place to me. Is the initial search really part of the framework lifecycle?

How does it work when the initial search query is disabled?

Reply:
Without the initial search the components load up empty and I can't imagine a use case where someone would want that. Normally the parent SearchInterface component would perform the initial search once all components were initialized. The headlessLoader is taking those responsibilities from it.

Reply:
> ... and I can't imagine a use case where someone would want that.

I saw it used many times, mostly for intranet use cases, to provide a Google-like experience where no results are shown until the user types in a query.

Reply:
Ah ok I see! Well I think we can add this kind of option later. I don't believe it's currently available with quantic as it is now and I'm more focused on feature parity + sibling enablement for this story.